### ISSUE TITLE
Support for Request Contexts

### ISSUE THREAD

**Issue by harmonherring** [KnutZuidema/golio#64]
It doesn't look like any of these requests support caller-defined contexts. @KnutZuidema would you support it if I opened a PR that adds `...WithContext` functions for all existing API requests?

For example, I'd add `GetByAccountIDWithContext(ctx context.Context, ...)`, `GetByPUUIDWithContext(ctx context.Context, ...)` etc etc for summoner routes.

I'm thinking I'd bubble that context down to a [duplicate](https://github.com/KnutZuidema/golio/blob/master/internal/client.go#L122) `Client.DoRequestWithContext(ctx context.Context, ...)` that adds a context to the request with `request.WithContext`

**Comment by KnutZuidema** [comment]
@harmonherring makes sense to add these. However, instead of creating new methods next to the existing ones, I would bump the major version of the library instead and replace/update the existing methods.

**Comment by happsie** [comment]
@KnutZuidema in regards to moving to a v2 API, is that something you feel like you want to do or could we help with this? Having a request context can be very good since this library is interacting with third party. Don't want to waste resources/potential rate limiting. 

Just wanted to ask since I feel that it will be a pretty big bump in the library and maybe you'd like control of that

**Comment by happsie** [comment]
Adding to this. Maybe it would be possible to add the option pattern to solve this as well without breaking the backwards compatability of the library.
